futures-util/src/sink/unfold.rs

```diff
@@ -0,0 +1,83 @@
+use core::{future::Future, pin::Pin};
+use futures_core::task::{Context, Poll};
+use futures_sink::Sink;
+use pin_project::pin_project;
+
+/// Sink for the [`unfold`] function.
+#[pin_project]
+#[derive(Debug)]
+#[must_use = "sinks do nothing unless polled"]
+pub struct Unfold<T, F, R> {
+    state: Option<T>,
+    function: F,
+    #[pin]
+    future: Option<R>,
```

Comment:
`state` and `future` are never present at the same time, so maybe the use of an enum could reduce the size of the struct?

Reply:
I would accept a patch to reduce the size of `unfold`.